polkadot-host-spec/c02-state.tm

```diff
@@ -476,42 +476,193 @@
+  <reference|table-changes-trie-key-types>. inside the trie. While the
+  overall structure of each keys is almost identical, each mapping has it own
+  prefix and could therefore be considered a <with|font-shape|italic|varying
+  datatype> (as defined in <reference|defn-varrying-data-type>). In contrast
+  to a <with|font-shape|italic|varying datatype> however, the type
+  identifiers do not describes the type of the key following it, but the
+  corresponding value of that key. The format and use of these values are
+  specified in more detail in their corresponding definitions
+  <reference|defn-storage-key-to-extrinsics>,
+  <reference|defn-storage-key-to-blocks> and
+  <reference|defn-storage-key-to-child-tries> respectivly.<htab|5mm>
+
+  <\big-table>
+    <tabular|<tformat|<cwith|2|2|1|-1|cell-bborder|0ln>|<cwith|3|3|1|-1|cell-tborder|1ln>|<cwith|3|3|1|-1|cell-bborder|0ln>|<cwith|4|4|1|-1|cell-tborder|1ln>|<cwith|4|4|1|-1|cell-bborder|1ln>|<cwith|4|4|1|1|cell-lborder|0ln>|<cwith|4|4|3|3|cell-rborder|0ln>|<cwith|1|1|1|-1|cell-tborder|1ln>|<cwith|1|1|1|-1|cell-bborder|1ln>|<cwith|2|2|1|-1|cell-tborder|1ln>|<cwith|1|1|1|1|cell-lborder|0ln>|<cwith|1|1|3|3|cell-rborder|0ln>|<table|<row|<cell|<strong|Id>>|<cell|<strong|Description>>|<cell|<strong|Followed
+    by>>>|<row|<cell|1>|<cell|Mapping between storage key and extrinsics
+    (<reference|defn-storage-key-to-extrinsics>)>|<cell|<em|KeyIndex>
+    (<reference|defn-key-index>)>>|<row|<cell|2>|<cell|Mapping between
+    storage key and blocks (<reference|defn-storage-key-to-blocks>)>|<cell|<em|KeyIndex>
+    (<reference|defn-key-index>)>>|<row|<cell|3>|<cell|Mapping between
+    storage key and child changes Trie (<reference|defn-storage-key-to-child-tries>)>|<cell|<em|KeyIndex>
+    (<reference|defn-key-index>)>>>>>
+
+    \;
+  <|big-table>
+    <label|table-changes-trie-key-types>Possible types of keys of mappings in
+    the Changes Trie
+  </big-table>
+
+  <strong|Note>: Unlike the default encoding for varying data types, this
+  structure starts its indexing at <verbatim|1>.
+
+  <\definition>
+    <label|defn-key-index>The <strong|Key Index> is a tuple containing the
+    following information:
+
+    <\big-table|<tabular|<tformat|<cwith|2|2|1|1|cell-lborder|0ln>|<cwith|2|2|3|3|cell-rborder|0ln>|<cwith|3|3|1|-1|cell-tborder|1ln>|<cwith|2|2|1|-1|cell-bborder|1ln>|<cwith|3|3|1|-1|cell-bborder|1ln>|<cwith|3|3|1|1|cell-lborder|0ln>|<cwith|3|3|3|3|cell-rborder|0ln>|<cwith|1|1|1|-1|cell-tborder|1ln>|<cwith|1|1|1|-1|cell-bborder|1ln>|<cwith|2|2|1|-1|cell-tborder|1ln>|<cwith|1|1|1|1|cell-lborder|0ln>|<cwith|1|1|3|3|cell-rborder|0ln>|<table|<row|<cell|<strong|Name>>|<cell|<strong|Description>>|<cell|<strong|Type>>>|<row|<cell|Block>|<cell|Block
+    number at which this key has been inserted in the trie>|<cell|Unsigned
+    32-bit integer>>|<row|<cell|Key>|<cell|The changed key>|<cell|Byte
+    array>>>>>>
+      Key structure inserted in the Changes Trie
+    </big-table>
+  </definition>
+
```

Comment:
For all these 3 definitions I would define a formal function (give it a name) define its  domain and range type and it is criteria (similar the childrenbitmap for example). It will be less confusing.